File: java/practice10.java
interface I {
    void check();
}

abstract class B {
    int s=10;
    B(B obj) {
     
    }
}

class A extends B implements I {
    int b;
    public void check() {
        System.out.println("Class A check function");
    }
    
}

public class practice10 {
    public static void main(String args[]) {
        I i1 = new A();
        i1.check();
    }
}

File: c++ prog/practice10.cpp
#include <iostream>
using namespace std;
class a
{
    void check1();

protected:
    int x;
    void check()
    {
        cout << "the check function was invoked";
    }
};
class b : protected a
{
public:
    void check2()
    {
        a obj1;
        check();
        x = 12;
    }
};
int main()
{
    b obj;
    obj.check2();
    return 0;
}